test/martha/martha_v3.test.js

```diff
@@ -279,30 +536,86 @@ test.serial('martha_v3 calls bond Bond with the "fence" provider when the Data O
+    const result = response.send.lastCall.args[0];
+    t.true(getJsonFromApiStub.calledOnce); // Bond was not called to get SA key
+    t.deepEqual(Object.assign({}, result), jadeDosMarthaResult(null));
+    t.falsy(result.googleServiceAccount);
+    t.is(response.statusCode, 200);
+});
+
+test.serial('martha_v3 parses Gen3 CRDC response correctly', async (t) => {
+    getJsonFromApiStub.onFirstCall().resolves(gen3CrdcResponse);
     const response = mockResponse();
     await marthaV3(mockRequest({ body: { 'url': 'drs://jade.datarepo-dev.broadinstitute.org/abc' } }), response);
     const result = response.send.lastCall.args[0];
     t.true(getJsonFromApiStub.calledOnce); // Bond was not called to get SA key
-    t.deepEqual(Object.assign({}, result), fullExpectedResult(null));
+    t.deepEqual(Object.assign({}, result), gen3CrdcDrsMarthaResult(null));
+    t.falsy(result.googleServiceAccount);
+    t.is(response.statusCode, 200);
+});
+
+test.serial('martha_v3 parses BDC response correctly', async (t) => {
+    getJsonFromApiStub.onFirstCall().resolves(bdcDrsResponse);
+    const response = mockResponse();
+    await marthaV3(mockRequest({ body: { 'url': 'dos://jade.datarepo-dev.broadinstitute.org/abc' } }), response);
+    const result = response.send.lastCall.args[0];
+    t.true(getJsonFromApiStub.calledOnce); // Bond was not called to get SA key
+    t.deepEqual(Object.assign({}, result), bdcDrsMarthaResult);
+    t.falsy(result.googleServiceAccount);
+    t.is(response.statusCode, 200);
+});
+
+test.serial('martha_v3 parses Anvil response correctly', async (t) => {
+    getJsonFromApiStub.onFirstCall().resolves(anvilDrsResponse);
+    const response = mockResponse();
+    await marthaV3(mockRequest({ body: { 'url': 'dos://jade.datarepo-dev.broadinstitute.org/abc' } }), response);
+    const result = response.send.lastCall.args[0];
+    t.true(getJsonFromApiStub.calledOnce); // Bond was not called to get SA key
+    t.deepEqual(Object.assign({}, result), anvilDrsMarthaResult(null));
+    t.falsy(result.googleServiceAccount);
+    t.is(response.statusCode, 200);
+});
+
+test.serial('martha_v3 parses Kids First response correctly', async (t) => {
+    getJsonFromApiStub.onFirstCall().resolves(kidsFirstDrsResponse);
+    const response = mockResponse();
+    await marthaV3(mockRequest({ body: { 'url': 'dos://jade.datarepo-dev.broadinstitute.org/abc' } }), response);
+    const result = response.send.lastCall.args[0];
+    t.true(getJsonFromApiStub.calledOnce); // Bond was not called to get SA key
+    t.deepEqual(Object.assign({}, result), kidsFirstDrsMarthaResult(null));
+    t.falsy(result.googleServiceAccount);
+    t.is(response.statusCode, 200);
+});
+
+test.serial('martha_v3 parses HCA response correctly', async (t) => {
+    getJsonFromApiStub.onFirstCall().resolves(hcaDosResponse);
+    const response = mockResponse();
+    await marthaV3(mockRequest({ body: { 'url': 'dos://jade.datarepo-dev.broadinstitute.org/abc' } }), response);
```

Comment:
The host name in the request url should end with HCA, for example `dos://someservice.humancellatlas.org/abc`. I think this is the reason the response has most of the values as null. Since the request has JDR host, it tries to parse the response using DRS v1 schema instead of the DOS schema resulting in null values for almost everything. 

Similarly the test for Anvil, Kids first, BDC and Gen3 should also have their respective hostnames in the request

Reply:
As Saloni said, hopefully this explains why you have the comment above `hcaDosMarthaResult`: `Quite a grim result, but could be working as expected?`

Also, I think we might want to add another part to each of these tests.  `determineDrsType` determines what we should be sending to our 3rd parties, what the URL and parameters look like.  As a part of testing the boundaries, we may want to test that as a part of these tests.  We should probably test the arguments that are being sent to  `getJsonFromApiStub.onFirstCall()`.

Reply:
Nice, when updated to the URL Saloni suggested the response is reasonable. This tells me the test is checking something valuable, because there is evidently important special handling on the URL.

Next question (also noted in a code comment), why is `size` a string in HCA but an int elsewhere?

[I recognize that the goal of this story is to capture existing behavior, but I think apparent inconsistency warrants examination]

Reply:
Huh, that's interesting. What happens when you keep the size as an Int in the response [here](https://github.com/broadinstitute/martha/blob/51df8c8837020b096bc40cfc9370542d29b10b0a/test/martha/martha_v3.test.js#L365)? Does the test fail?

Reply:
I think it's because of [this](https://github.com/broadinstitute/martha/blob/630f1b7af769f1db08d78d6798c3d6ed0fad3d3b/martha/martha_v3.js#L76) line of code. So it looks like if the `size` in the result that you get back from external DRS server is an `Int`, `size` in the final martha_v3 response will be an `Int`. Same goes for it being a `String`.

Reply:
Yes, the test fails when the types do not match.

I think it's not a good design to vary the output type based on the input, would you agree if I modified the code to always cast to int?

Reply:
Yes I agree that we should not vary the output based on input. I think you can change the code since you found a bug.

Reply:
Cool, done.